### PR TITLE
Fft rewrite

### LINKED ISSUE #181: Replace fftw interface with Eigen fftw
would allow us to use eigen internal implementation without fftw library 
and otherwise we could use fftw or MKL as a backend. 

https://eigen.tuxfamily.org/dox/unsupported/group__FFT__Module.html

https://gitlab.math.ethz.ch/NumCSE/NumCSE/blob/43c3152296a97f4a927cd82fe3c19ac0c64537e4/CMake/Macros/set_eigen_fft_backend.cmake

### PR DESCRIPTION
**Pull request by JensWehner**
implements https://github.com/votca/tools/issues/181

Part of https://github.com/votca/votca/pull/219

Fix #181 
Related to https://github.com/votca/csg/issues/104

### CODE REVIEW

File: include/votca/tools/eigen.h [hunk 1]
```diff
@@ -27,6 +27,11 @@
 #define EIGEN_USE_MKL_ALL
 #endif
 
+#ifndef NOFFTW
+#include <fftw3.h>
```
Comment (junghans) [inline include/votca/tools/eigen.h]: I don't think this line is needed as `/usr/include/eigen3/unsupported/Eigen/FFT` includes `fftw3.h` already.

Comment (JensWehner) [reply include/votca/tools/eigen.h]: ahh good point.

File: include/votca/tools/eigen.h [hunk 2]
```diff
@@ -45,6 +49,7 @@
 #endif
 #include <Eigen/Eigen>
 #include <unsupported/Eigen/CXX11/Tensor>
+#include <unsupported/Eigen/FFT>
```
Comment (junghans) [inline include/votca/tools/eigen.h]: Is there a reason to always include this? Why not just the correlate class?

Comment (JensWehner) [reply include/votca/tools/eigen.h]: I thought it is maybe a good idea to make it available to all of votca, as people might want to use this functionality.

Comment (JensWehner) [reply include/votca/tools/eigen.h]: because FFT could come up quite quickly in XTP as well

Comment (JensWehner) [reply include/votca/tools/eigen.h]: but putting it into correlate would probably be enough

Comment (JensWehner) [reply include/votca/tools/eigen.h]: I think FFT could be useful for further development and it is pretty well namespaced so I actually would make it available in all of votca

Comment (JensWehner) [reply include/votca/tools/eigen.h]: what do you think?

Comment (junghans) [reply include/votca/tools/eigen.h]: No preference just wondering.

**Pull request merged**